### PR TITLE
fix podman generate kube with HostAliases

### PR DESCRIPTION
**Pull request by zhangguanzhang**
Fixes: https://github.com/containers/podman/issues/7468
Signed-off-by: zhangguanzhang <zhangguanzhang@qq.com>

### CODE REVIEW

File: libpod/kube.go [hunk 1]
```diff
@@ -48,19 +48,33 @@ func (p *Pod) GenerateForKube() (*v1.Pod, []v1.ServicePort, error) {
...
 			return nil, servicePorts, err
 		}
 		servicePorts = containerPortsToServicePorts(ports)
 	}
 	pod, err := p.podWithContainers(allContainers, ports)
+	if err != nil {
+		return nil, servicePorts, err
+	}
+	pod.Spec.HostAliases = extraHost
 	return pod, servicePorts, err
```
Comment (rhatdan) [inline libpod/kube.go:78]: Can you change this to return nil, rather then err, to make it clearer to future  developers that this is success.

**Comment by rhatdan** [comment]
@mheon @vrothberg @haircommander @giuseppe @mrunalp @baude PTAL

File: libpod/kube.go (hunk 1 above)
Comment (zhangguanzhang) [reply libpod/kube.go:78]: servicePorts []v1.ServicePort   //nolint
nolint, so I should'nt use nil, and line 47 
```
return nil, servicePorts, errors.Erro....
```

Comment (zhangguanzhang) [reply libpod/kube.go]: I got it wrong🤣, now it's handled

**Comment by TomSweeneyRedHat** [comment]
LGTM
and happy green test buttons.  How did that happen?

**Comment by zhangguanzhang** [comment]
I think the latest code of other people is also green in ci after merging, which may be fixed by a previous submission

**Comment by rhatdan** [comment]
We need someone with a better handle on Kubernetes to review this.
@mheon @vrothberg @haircommander @giuseppe @mrunalp @baude @umohnani8  PTAL

**Comment by haircommander** [comment]
LGTM

**Comment by umohnani8** [comment]
LGTM

**Pull request merged**